### DISCUSSION TITLE
Door ajar warning

### DISCUSSION THREAD

**Discussion by rayr007** [thoukydides/homebridge-homeconnect#106] [Ideas]
Would it be possible to map door ajar to the door open in homekit? Currently, I turned off notifications for door open on my fridge as it's too much noise every time somebody opens the door. However, if homeconnect exposes the door left open notification, having that pop up as "door open" would be nice. 

I know I could do some sort of automation on the door open/close notifications. Although that is cumbersome and then requires some external notification service from what I can tell. I figured I'd ask if this other idea has any potential...

**Comment by thoukydides** [comment]
Home Connect events are essentially stateless. There is no way to read from the API what the current state is, unless that is also provided via a non-event mechanism. This is why this plugin maps them to Stateless Programmable Switches that can trigger automations.

Attempting to track the appliance state from the events is likely to end up with incorrect results much of the time, especially after any homebridge reset or temporary loss of communication with the Home Connect cloud/appliance.

This is why I only added the Door Service after the Home Connect API introduced an explicit Door State that could be read at any time, rather than just the Door State Change events. (I am not sure how it behaves on an appliance with multiple doors, e.g. a FridgeFreezer...)

Additionally, this would require special case handling for Fridge/Freezer/FridgeFreezer appliances, treating their doors differently to those of other appliances like Ovens and Dishwashers. I do not own a Home Connect Fridge/Freezer/FridgeFreezer so would be unable to test that code. Hence, this is not something that I want to do. I made a special case for Hood appliances where their functionality actually mapped cleanly to less generic HomeKit services and characteristics, such as fans and lightbulbs. I am not willing to do that where it doesn't improve the HomeKit mapping... especially for appliances that I do not own.

**Comment by rayr007** [comment]
Very understandable. Hopefully both  home connect and homekit continue to improve allowing for additional feature mapping.

FYI: I have a bosch fridge and the way it works (for mine at least) is that both the fridge and freezer trigger a single door open. So it doesn't matter which door you open/close, the same event is fired. It also means you don't know which door it was. I don't see this changing as I believe the home connect app behaves the same; my guess is that the door open sensors in the appliance are both wired to the same circuit...

The home connect app does do an alert when any door is left ajar. And I see 3 buttons for mine. Unfortunately both the home and eve apps show all the buttons/switches as named "Fridge/Freezer" so I have no idea what is what. :(

**Comment by thoukydides** [comment]
On a Fridge/Freezer appliance the three Stateless Programmable Switches (which the Home app calls "buttons") are, in order:
1. Freezer Door Alarm
2. Refrigerator Door Alarm
3. Freezer Temperature Alarm

The Home app should use those numbers.

The Eve app displays the name when used on an iPhone/iPad running iOS 15.x or earlier. The use of the appliance/accessory name for every button/switch is new behaviour that Apple have introduced in iOS 16. Please refer to https://github.com/thoukydides/homebridge-homeconnect/discussions/102 and https://github.com/homebridge/homebridge/issues/3210.

Home Connect Fridge/Freezer appliances can clearly distinguish between the fridge and freezer door states since the API generates different alarm events for them. It is just poor API design that they only report a single door position, which is whichever changed state last.
